docs/src/snap/howto/install/install-offline.md

```diff
@@ -0,0 +1,374 @@
+
+Test the connectivity:
+
+```
+export https_proxy=http://squid.internal:3128
+curl -v https://registry-1.docker.io 
+```
+<!-- TODO: 404 on curl and https://registry-1.docker.io/v2 unauthorized -->
+
+Please refer to the next section `images` on how to use the HTTP proxy
+to allow limited access to image registries.
+
+## Prep 3: Images
+
+All workloads in a Kubernetes cluster are running as an OCI image.
+Kubernetes needs to be able to fetch these images and load them
+into the container runtime, in order to run workloads.
+For a Canonical Kubernetes deployment, you will need to fetch the images
+used by its features (network, dns, etc) as well as any images that are
+needed to run your workloads.
+
+The following options are presented in the order of
+increasing complexity of implementation.
+You may also find it helpful to combine these options for your scenario.
+
+If you already have a running cluster,
+you can list the images in use by running:
+  
+```bash
+sudo k8s kubectl get node -o template='{{ range .items }}{{ .metadata.name }}\
+{{":"}}{{ range .status.images }}{{ "\n- " }}{{ index .names 1 }}\
+{{ end }}{{"\n"}}{{ end }}'
+```
+
+### Images Option A: via an HTTP proxy
+
+In many cases, the nodes of the airgap deployment may not have direct access to
+upstream registries, but can reach them through the
+[use of an HTTP proxy][proxy].
+
+To configure the proxy, edit `/etc/squid/squid.conf` and set the appropriate
+allowed networks and domains. 
+
+Upon changes to the proxy configuration,
+restart both the squid proxy and the k8s snap with:
+
+```bash
+sudo systemctl restart squid
+```
+  
+```bash
+sudo snap restart k8s
+```
+
+### Images Option B: private registry mirror
+
+In case regulations and/or network constraints do not allow the cluster nodes
+to access any upstream image registry,
+it is typical to deploy a private registry mirror.
+This is an image registry service that contains all the required OCI Images
+(e.g. [registry](https://distribution.github.io/distribution/),
+[Harbor](https://goharbor.io/) or any other OCI registry) and
+is reachable from all cluster nodes.
+
+This requires three steps:
+
+1. Deploy and secure the registry service. This is out of scope for this
+   document, please follow the instructions for the registry
+   that you want to deploy.
+2. Load all images from the upstream source and push to your registry mirror.
+3. Configure the Canonical Kubernetes container runtime (`containerd`) to load 
+   images from
+   the private registry mirror instead of the upstream source. This will be
+   described in the installation section.
+
+In order to load images into the private registry, you need a machine with
+access to both the upstream registry (e.g. `docker.io`) and the internal one.
+Loading the images is possible with `docker` or `ctr`.
+
+For the examples below we assume that a private registry mirror is running at
+`10.100.100.100:5000`.
+
+#### Load images with ctr
```

Comment:
for images, we should point people to use `regsync` instead.

consider adding links to:
- https://github.com/canonical/k8s-snap/blob/332ce3032b1cc811a671a1a4f5c647dcdd3d4175/build-scripts/hack/upstream-images.yaml, with instructions to adjust the target for their own registry mirror
- https://github.com/canonical/k8s-snap/blob/5b35ff89cce6dfa7c2b226cd7423c4630cf48756/build-scripts/hack/sync-images.sh to sync the images

in the past we referred people to `ctr` and `docker` for lack of a better alternative, which we now have and can fully control/version

Reply:
Ah, super helpful. Thanks Angelos!